include/nlohmann/detail/macro_scope.hpp

```diff
@@ -98,14 +98,39 @@
 #endif
 
 #ifndef JSON_HAS_THREE_WAY_COMPARISON
-    #if defined(__cpp_lib_three_way_comparison) && __cpp_lib_three_way_comparison >= 201907L \
-        && defined(__cpp_impl_three_way_comparison)&& __cpp_impl_three_way_comparison >= 201907L
-        #define JSON_HAS_THREE_WAY_COMPARISON 1
-    #else
-        #define JSON_HAS_THREE_WAY_COMPARISON 0
+    #if defined(__cpp_impl_three_way_comparison) && __cpp_impl_three_way_comparison >= 201907L \
+        && defined(__has_include)
+        #if __has_include(<compare>)
+            #include <compare>
+            #if defined(__cpp_lib_three_way_comparison) && __cpp_lib_three_way_comparison >= 201907L
+                #define JSON_HAS_THREE_WAY_COMPARISON 1
```

Comment:
`#if __has_include(<compare>)` needs to be on its own. If merged with the line above, MSVC's preprocessor isn't happy about the angle brackets.